src/apify/scrapy/middlewares/apify_proxy.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+from base64 import b64encode
+from typing import TYPE_CHECKING
+from urllib.parse import ParseResult, unquote, urlparse
+
+from scrapy.core.downloader.handlers.http11 import TunnelError
+from scrapy.exceptions import NotConfigured
+from scrapy.utils.python import to_bytes
+
+from apify import Actor, ProxyConfiguration
+
+if TYPE_CHECKING:
+    from scrapy import Request, Spider
+    from scrapy.crawler import Crawler
+
+
+class ApifyHttpProxyMiddleware:
+    """Apify HTTP proxy middleware for Scrapy.
+
+    This middleware enhances request processing by adding a 'proxy' field to the request's meta and an
+    authentication header. It draws inspiration from the Scrapy default `HttpProxyMiddleware`. The proxy URL is
```

Comment:
Sorry to be nitpicky, but in what sense is that Scrapy middleware default?

Reply:
Oh, I see, thanks. Then I'd probably reword this along the lines of "It draws inspiration from the `HttpProxyMiddleware` included by default in Scrapy projects". What do you think?

Reply:
Sure, done